### DISCUSSION TITLE
color picker for scene

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [zauberzeug/nicegui#1231] [Q&A] [answered]
### Question

https://github.com/WolfgangFahl/nicescad/issues/10 calls for a color picker see https://ochafik.com/openscad2 hat has one but http://nicescad.bitplan.com/ currently does not. How could the color of a scene be modified?

**Comment by rodja** [comment]
Have you seen https://nicegui.io/documentation/color_picker?

**Reply by WolfgangFahl** [reply]
Thx - i new about this.

**Comment by falkoschindler** [comment, marked as answer]
And what is "the color of a scene"?
To change the color of an object you can use the `.material('#ff0000')` method.

**Reply by WolfgangFahl** [reply]
The stl object is what i'd like to have the correct colors. Color handling of STL seems to be a known/open issue.

The original Viewscad code elaborately analyzed the stl file to allow to click individual objects which would also allow handling colors. I am not sure whether I intend to go that route (yet ...). For the time being i am happy with offering the option to change material color just as https://ochafik.com/openscad2/ does.

**Comment by WolfgangFahl** [comment]
Thank you both! I have closed https://github.com/WolfgangFahl/nicescad/issues/10
![grafik](https://github.com/zauberzeug/nicegui/assets/1336221/205e8e6a-e75d-4e84-b2c1-7103165be9c4)

But there is the followup issue how to properly position and size the picker so that it looks more pleasant.
